### PR TITLE
feat: moves html lang options from check to commons

### PR DESCRIPTION
**Pull request by downzer0**
The aXe rule/check for valid HTML `lang` attributes had an options array that was being included three times not only making the `axe.js` file three times larger than it needed to be, but also made other apps that consume aXe at least that much larger as well.

I've moved the options array from the check to an `axe.commons.utils` file that returns the same and updated the check JavaScript itself to use `options` if present, otherwise fallback to the returned array of every possible language code.

Closes https://github.com/dequelabs/axe-core/issues/348.

### CODE REVIEW

**Comment by downzer0** [comment]
@dylanb @iandotkelly @marcysutton @WilcoFiers I'm working on tests, atm, but heads-up!

**Comment by downzer0** [comment]
@dylanb @iandotkelly @marcysutton @WilcoFiers Ready for your review.

File: lib/commons/utils/valid-langs.js [hunk 1]
```diff
@@ -0,0 +1,8123 @@
+/*global axe */
+
+axe.utils.validLangs = function () {
+	'use strict';
```
Comment (marcysutton) [inline lib/commons/utils/valid-langs.js]: I don't think we need `use strict` as Babel adds it for us...as an FYI

File: test/checks/language/valid-lang.js [hunk 2]
```diff
@@ -43,13 +43,12 @@ describe('valid-lang', function () {
 			assert.deepEqual(checkContext._data, ['lang="en-FOO"']);
 		});
 
-		it('should return true (and not throw) when given no options', function () {
+		it('should return false (and not throw) when given no options', function () {
```
Comment (marcysutton) [inline test/checks/language/valid-lang.js]: How come the return values changed here?

Comment (downzer0) [inline test/checks/language/valid-lang.js]: It's related to the `none` in the check. It's a bit backwards. Previously, there were `options` or `[]`, but now we've got `(options ? options : axe.commons.utils.validLangs)` - so either `null` or some, but never empty.

This is my best explanation. It's kind of hard to wrap my head around the tests that are true, that return false :)

File: lib/commons/utils/valid-langs.js (hunk 1 above)
Comment (downzer0) [inline lib/commons/utils/valid-langs.js]: Some of the other utils have it, but happy to remove if you advise.

File: test/checks/language/valid-lang.js (hunk 2 above)
Comment (dylanb) [inline test/checks/language/valid-lang.js]: Add a test that tests to make sure it returns true if passed an empty options array

**Comment by dylanb** [comment]
@downzer0 - as mentioned in the call - the list of languages is built by the `grunt langs` command, so we need to make sure that build task still works with this new implementation

File: test/checks/language/valid-lang.js (hunk 2 above)
Comment (downzer0) [inline test/checks/language/valid-lang.js]: @dylanb doesn't the test on line 37 do that?

**Comment by downzer0** [comment]
@dylanb Everything seems to build OK, and when I manually bring in the compiled `axe.js` file into our rulebuilder, things work there too, in addition to the filesize being 2/3 smaller.

Should we remove the Grunt task?

**Comment by dylanb** [comment]
No, the grunt task does not get called automatically, it is there to build the list should the languages change - now admittedly, this happens very infrequently, but calling it periodically will automatically update for changes whereas maintaining that manually is almost impossible due to the size.

So the Grunt task should be changed to generate the code in the commons from a template (or something equivalent)

**Comment by downzer0** [comment]
@dylanb I've made an update which alters the Gruntfile location to the new util file.

I originally had this generate a JSON file and then tried `require`-ing it in the check, but require is undefined. I opted to just re-build the entire check file with the languages array. I had to disable the JSHint double/single quote check for this file because JSON.stringify makes valid JSON (which has double-quotes) but our rules want single.

I've tested the build process and tested the output axe.js in our rulebuilder.

**Comment by marcysutton** [comment]
So this task for running langs...we should document it better. There is a mention of `valid-lang` in [doc/developer-guide.md](https://github.com/dequelabs/axe-core/blob/develop/doc/developer-guide.md) we need to update. 

I'm also wondering if we should register a task at the bottom of the Gruntfile to make this more obvious.

**Comment by downzer0** [comment]
@marcysutton Good catch, with the documentation - I'll get it updated. My only hesitation with the registered task is that, based on @dylanb's comment, it doesn't sound like this is used much. Also, you _can_ run it with `grunt langs`, it just isn't exposed as easily. I am happy to expose it, but will defer to you and Dylan on this.

**Comment by marcysutton** [comment]
I didn't even know it was there, since there is another variable called `langs`. It is super confusing, so we should do something to document it at least.

**Comment by downzer0** [comment]
Just looked - the docs don't need an update. I will get the Grunt task documented though.

**Comment by downzer0** [comment]
We will document the Grunt tasks in a separate PR. Marcy is creating an issue for it.

**Comment by marcysutton** [comment]
Looks good! Circle just needed a rebuild.
